app/src/main/java/com/firebase/uidemo/auth/AccountLinkActivity.java

```diff
@@ -0,0 +1,197 @@
+package com.firebase.uidemo.auth;
```

Comment:
Decided to do a new sample here since this is an "advanced" feature and I don't want to clutter the main sample.

Reply:
If we were going your way with this feature (service, data callbacks, etc) then I'd agree with merging the samples.  However in this case since you have to write extra code and really understand the merge-failure-upgrade flow I wanted to give it dedicated UI and keep the code separate so as not so confuse users of the 'basic' AuthUI flow with all the extra stuff.

If you try out the sample, you can see how it makes the various states extra clear.

Reply:
Gotya, that totally makes sense so 👍.